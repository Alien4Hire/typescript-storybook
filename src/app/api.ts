import {
  BaseQueryFn,
  FetchArgs,
  FetchBaseQueryError,
} from '@reduxjs/toolkit/query';

import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { logout, getCaTokenCookie } from './reducers/login/loginSlice';
import { setToastNotification } from './reducers/appSlice/appSlice';

export const baseUrl = process.env.REACT_APP_API || '/api/v1';

declare module '@reduxjs/toolkit/query' {
  interface FetchBaseQueryError {
    status: number;
    data:
    | {
      id?: string;
      location?: string;
      message: string;
      path?: string;
      status?: number;
      type?: string;
    }[]
    | {
      id?: string;
      location?: string;
      message: string;
      path?: string;
      status?: number;
      type?: string;
    };
  }
}

const baseQuery = fetchBaseQuery({
  baseUrl,
  prepareHeaders: (headers, { getState }) => {
    const token = getCaTokenCookie();
    if (token) {
      headers.set('authorization', `Bearer ${token}`);
    }
    headers.set('Content-Type', 'application/json;charset=UTF-8');
    return headers;
  },
});

const baseQueryWithErrorHandling: BaseQueryFn<
  string | FetchArgs,
  unknown,
  FetchBaseQueryError
> = async (args, api, extraOptions) => {
  let result = await baseQuery(args, api, extraOptions);
  if (result.error) {
    if (result.error.status === 401) {
      api.dispatch(logout());
      /**
       * Instead of logging out when the token is invalid, it is possible to refresh the token here.
       */
      // try to get a new token
      // const refreshResult = await baseQuery('/refreshToken', api, extraOptions);
      // if (refreshResult.data) {
      //   // store the new token
      //   api.dispatch(setNewToken(refreshResult.data));
      //   // retry the initial query
      //   result = await baseQuery(args, api, extraOptions);
      // } else {
      //   api.dispatch(logout());
      // }
    } else {
      const message = Array.isArray(result.error?.data)
        ? result.error?.data[0]?.message
        : result.error?.data?.message;
      api.dispatch(
        setToastNotification({
          type: 'error',
          open: true,
          message,
        })
      );
    }
  }
  return result;
};

export const apiBase = createApi({
  reducerPath: 'api',
  baseQuery: baseQueryWithErrorHandling,
  //List of tag associated with our data endpoints
  tagTypes: [
    'Templates',
    'ActivitySettings',
    'TemplateViewTypes',
    'ProductModels',
    'LearningObjectives',
    'Taxonomies',
    'Activities',
    'Lessons',
    'Content',
  ],
  refetchOnFocus: true,
  refetchOnReconnect: true,
  refetchOnMountOrArgChange: true,
  endpoints: () => ({}),
});
