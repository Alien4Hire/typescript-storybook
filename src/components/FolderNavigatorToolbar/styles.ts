import { Box, Button } from '@mui/material';

import { styled } from '@mui/material/styles';

export const Divider = styled('div')(({ theme }) => ({
  borderRight: `1px solid ${theme.palette.gray.medium}`,
  height: '34px',
  margin: 'auto',
}));

export const Label = styled('div')(({ theme }) => ({
  fontWeight: '600',
  fontSize: '15px',
  letterSpacing: '0.4px',
  color: theme.palette.gray.veryDark,
  fontFamily: "'Source Sans Pro', sans-serif",
}));

export const Wrapper = styled('div')(({ theme }) => ({
  position: 'relative',
  background: theme.palette.gray.veryLight,
  paddingLeft: '56px',
}));

export const FolderForm = styled('div')`
  display: flex;
  align-items: center;
`;

export const FolderAdd = styled(Box, {
  shouldForwardProp: (prop: string) => prop !== 'active',
})(({ active, theme }) => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: 'auto',
  padding: '2px',
  backgroundColor: theme.palette?.gray.veryLight,
  display: 'none',
  ...(active && {
    display: 'flex',
    alignItems: 'center',
  }),
}));

export const CsvModalBackdrop = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f1f1f1',
  borderRadius: '16px',
  padding: '32px 32px 70px',
});

export const UploadListItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: '16px 0px',
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const UploadButton = styled(Button)({
  textTransform: 'none',
  width: 'fit-content',
  marginTop: '16px',
});

export const DownloadTemplateWrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  marginTop: '24px',
  cursor: 'pointer',
});
