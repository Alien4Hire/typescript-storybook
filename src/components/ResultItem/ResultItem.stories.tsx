import React from 'react';
import { useArgs } from '@storybook/client-api';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import ProviderWrapper from '../../util/ProviderWrapper';
import Component, { ResultItemPropsT } from './ResultItem';

export default {
  title: 'Component/ResultItem',
  component: Component,
  decorators: [
    (Story) => (
      <ProviderWrapper>
        <Story />
      </ProviderWrapper>
    ),
  ],
} as ComponentMeta<typeof Component>;

const Template: ComponentStory<typeof Component> = (args) => {
  const [local, updateArgs] = useArgs();

  const onCheck = () => {
    updateArgs({ checked: !local.checked });
    args.onCheck(args.contentId);
  };

  return <Component {...args} onCheck={onCheck} />;
};

export const ResultItem = Template.bind({});
ResultItem.args = {
  checked: false,
  checkedValue: 'test',
  activityName: 'Ch 1 LearningCurve Adaptive Quiz',
  contentId: 'XXXXXXXX',
  modifiedDate: 'xx/xx/xxxxx at xx:xx',
  addedToLibrary: false,
  onCheck: () => {},
  onAddResource: () => {},
} as ResultItemPropsT;
