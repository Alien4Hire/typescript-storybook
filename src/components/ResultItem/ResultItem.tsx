import { Box, Checkbox, Typography } from '@mui/material';
import { ResultItemT } from '../../app/reducers/search/searchSlice';
import { useState } from 'react';
import { MlDropdownMenuOptionPropsT } from '../DropdownMenu/DropdownMenu';
import FolderDropdown from '../FolderDropdown/FolderDropdown';
import {
  Container,
  LinkText,
  ResourceAction,
  StyledAddIcon,
  StyledAssignmentIcon,
  StyledCheckCircleIcon,
} from './ResultItem.styles';

export type ResultItemPropsT = ResultItemT & {
  onCheck: (id: string) => void;
  onAddResource?: (id: string, folder: MlDropdownMenuOptionPropsT) => void;
};

const ResultItem = ({
  checkedValue,
  checked,
  onCheck,
  activityName,
  contentId,
  modifiedDate,
  onAddResource,
  addedToLibrary,
}: ResultItemPropsT) => {
  const subtitle = `ID: ${contentId} Last Modified: ${new Date(
    modifiedDate
  ).toLocaleString()}`;

  const handleActionClick = (folder: MlDropdownMenuOptionPropsT) => {
    !addedToLibrary && onAddResource?.(contentId, folder);
  };

  const onHandleCheck = () => {
    onCheck(contentId);
  };

  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <Container checked={checked}>
      <Checkbox
        value={checkedValue}
        checked={checked}
        onChange={onHandleCheck}
      />
      <Box display="flex" alignItems="center">
        <StyledAssignmentIcon />
        <Box ml="10px">
          <Typography
            sx={(theme) => ({
              color: theme.palette.aqua.main,
              mb: '7px',
            })}
          >
            {activityName}
          </Typography>
          <Typography
            sx={(theme) => ({
              color: theme.palette.gray.veryDark,
            })}
          >
            {subtitle}
          </Typography>
        </Box>
      </Box>
      <ResourceAction onClick={handleClick} role="button">
        {addedToLibrary ? (
          <>
            <StyledCheckCircleIcon />
            <LinkText>In Library</LinkText>
          </>
        ) : (
          <>
            <StyledAddIcon />
            <Typography
              variant="textSub"
              sx={(theme) => ({
                color: theme.palette.gray.veryDark,
                mt: '3px',
                lineHeight: '15px',
              })}
            >
              Add Resource
            </Typography>
          </>
        )}
      </ResourceAction>
      <FolderDropdown
        open={Boolean(anchorEl)}
        anchorEl={anchorEl}
        onItemClick={handleActionClick}
        onClose={handleClose}
      />
    </Container>
  );
};

export default ResultItem;
