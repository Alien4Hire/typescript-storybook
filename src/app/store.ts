import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { combineReducers } from '@reduxjs/toolkit';
import configReducer from './reducers/config/configSlice';
import loginReducer from './reducers/login/loginSlice';
import activitySettingsReducer from './reducers/activitySettings/activitySettingsSlice';
import lessonsReducer from './reducers/lessons/lessonsSlice';
import searchReducer from './reducers/search/searchSlice';
import resourceFilterReducer from './reducers/resourcesFilter/resourcesFilterSlice';
import templateReducer from './reducers/templates/templateSlice';
import appReducer from './reducers/appSlice/appSlice';
import templateViewTypesReducer from './reducers/templateViewTypes/templateViewTypesSlice';
import targetFolderSideBarReducer from './reducers/targetFolderSideBar/targetFolderSideBarSlice';

import { setupListeners } from '@reduxjs/toolkit/query/react';
import { apiBase } from './api';

const rootReducer = combineReducers({
  config: configReducer,
  login: loginReducer,
  activitySettings: activitySettingsReducer,
  lessons: lessonsReducer,
  templates: templateReducer,
  search: searchReducer,
  resourcesFilter: resourceFilterReducer,
  app: appReducer,
  templateViewTypes: templateViewTypesReducer,
  targetFolderSideBar: targetFolderSideBarReducer,
  [apiBase.reducerPath]: apiBase.reducer,
});

// Use also for store setup in jest unit tests
// Use-case: unit test needs preloadedState
// Example: client/src/components/Pages/LoginPage/LoginPage.test.tsx
export const setupStore = (preloadedState: object = {}) => {
  return configureStore({
    reducer: rootReducer,
    devTools: process.env.NODE_ENV !== 'production',
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(apiBase.middleware),
    preloadedState,
  });
};

export const store = setupStore();

setupListeners(store.dispatch);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
