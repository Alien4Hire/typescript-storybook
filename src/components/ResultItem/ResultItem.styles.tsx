import { Box, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';
import CheckCircleIcon from '@mui/icons-material/CheckCircle';
import AssignmentIcon from '@mui/icons-material/Assignment';
import AddIcon from '@mui/icons-material/Add';

export const Container = styled(Box)(({ theme, ...props }) => ({
  display: 'grid',
  gridTemplateRows: '1fr',
  gridTemplateColumns: '45px 1fr 78px',
  alignItems: 'center',
  padding: '12px 20px',
  backgroundColor: props.checked
    ? theme.palette.aqua.veryLight
    : theme.palette.white.main,
  boxShadow: `0px -1px 0px ${theme.palette.gray.light}`,
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const ResourceAction = styled(Box)(() => ({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  cursor: 'pointer',
  alignSelf: 'center',
}));

export const LinkText = styled(Typography)(({ theme }) => ({
  textDecoration: 'underline',
  marginTop: '5px',
  color: theme.palette.aqua.main,
  fontSize: 12,
  lineHeight: '15px',
}));

export const StyledAssignmentIcon = styled(AssignmentIcon)(({ theme }) => ({
  fill: theme.palette.green.medium,
}));

export const StyledCheckCircleIcon = styled(CheckCircleIcon)(({ theme }) => ({
  fill: theme.palette.green.medium,
}));

export const StyledAddIcon = styled(AddIcon)(({ theme }) => ({
  fill: theme.palette.aqua.main,
}));
