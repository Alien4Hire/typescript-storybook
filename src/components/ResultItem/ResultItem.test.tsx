// import { fireEvent } from '@storybook/testing-library';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import { store } from '../../app/store';
import '../../setupTests';
import { theme } from '../../theme';
import ResultItem, { ResultItemPropsT } from './ResultItem';

const baseProps: ResultItemPropsT = {
  addedToLibrary: true,
  checked: false,
  checkedValue: 'testValue',
  contentId: 'ID',
  activityName: 'Title',
  modifiedDate: 'xx/xx/xxxx at xx/xx',
  onAddResource: jest.fn(),
  onCheck: jest.fn(),
};

const Component = (props: ResultItemPropsT) => (
  <Provider store={store}>
    <Router>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>
          <ResultItem {...props} />
        </ThemeProvider>
      </StyledEngineProvider>
    </Router>
  </Provider>
);

const renderComponent = (customProps: Partial<ResultItemPropsT> = {}) => {
  const props = { ...baseProps, ...customProps };

  return render(<Component {...props} />);
};

beforeAll(() => {
  window.getComputedStyle = jest.fn();
});

describe('<ResultItem />', () => {
  it('renders without crashing', () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });

  it('should render passed fields', () => {
    renderComponent();

    [baseProps.contentId, baseProps.activityName].forEach((item) => {
      screen.getByText(new RegExp(item));
    });
  });

  it('should render icon when resource was added', () => {
    renderComponent();

    expect(screen.getByText(/In Library/)).toBeInTheDocument();
  });

  it('should render "Add Resource button"', () => {
    renderComponent({ addedToLibrary: false });

    const addResourceButton = screen.getByText(/Add Resource/);
    expect(addResourceButton).toBeInTheDocument();
  });
});
