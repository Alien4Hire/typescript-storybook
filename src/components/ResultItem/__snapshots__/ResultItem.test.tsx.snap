// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultItem /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c1::-moz-focus-inner {
  border-style: none;
}

.c1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c9 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  color: #00758e;
  margin-bottom: 7px;
}

.c10 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  color: #383838;
}

.c13 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  margin-left: 10px;
}

.c2 {
  padding: 9px;
  border-radius: 50%;
}

.c4 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.c3 {
  color: rgba(0,0,0,0.6);
}

.c3:hover {
  background-color: rgba(0,117,142,0.04);
}

.c3.Mui-checked,
.c3.MuiCheckbox-indeterminate {
  color: #00758e;
}

.c3.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c0 {
  display: grid;
  grid-template-rows: 1fr;
  grid-template-columns: 45px 1fr 78px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 12px 20px;
  background-color: #ffffff;
  box-shadow: 0px -1px 0px #dddddd;
  border-bottom: 1px solid #dddddd;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c14 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin-top: 5px;
  color: #00758e;
  font-size: 12px;
  line-height: 15px;
}

.c7 {
  fill: #68b75f;
}

.c12 {
  fill: #68b75f;
}

@media print {
  .c1 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c3:hover {
    background-color: transparent;
  }
}

<body>
    <div>
      <div
        class="c0 MuiBox-root"
      >
        <span
          class="c1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root c2 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root c3"
        >
          <input
            class="c4 PrivateSwitchBase-input"
            data-indeterminate="false"
            type="checkbox"
            value="testValue"
          />
          <svg
            aria-hidden="true"
            class="c5 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <div
          class="c6 MuiBox-root"
        >
          <svg
            aria-hidden="true"
            class="c5 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c7"
            data-testid="AssignmentIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
            />
          </svg>
          <div
            class="c8 MuiBox-root"
          >
            <p
              class="c9 MuiTypography-root MuiTypography-body1"
            >
              Title
            </p>
            <p
              class="c10 MuiTypography-root MuiTypography-body1"
            >
              ID: ID Last Modified: Invalid Date
            </p>
          </div>
        </div>
        <div
          class="c11 MuiBox-root"
          role="button"
        >
          <svg
            aria-hidden="true"
            class="c5 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c12"
            data-testid="CheckCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          <p
            class="c13 MuiTypography-root MuiTypography-body1 c14"
          >
            In Library
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": .c5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c1::-moz-focus-inner {
  border-style: none;
}

.c1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c9 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  color: #00758e;
  margin-bottom: 7px;
}

.c10 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  color: #383838;
}

.c13 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  margin-left: 10px;
}

.c2 {
  padding: 9px;
  border-radius: 50%;
}

.c4 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.c3 {
  color: rgba(0,0,0,0.6);
}

.c3:hover {
  background-color: rgba(0,117,142,0.04);
}

.c3.Mui-checked,
.c3.MuiCheckbox-indeterminate {
  color: #00758e;
}

.c3.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c0 {
  display: grid;
  grid-template-rows: 1fr;
  grid-template-columns: 45px 1fr 78px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 12px 20px;
  background-color: #ffffff;
  box-shadow: 0px -1px 0px #dddddd;
  border-bottom: 1px solid #dddddd;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c14 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin-top: 5px;
  color: #00758e;
  font-size: 12px;
  line-height: 15px;
}

.c7 {
  fill: #68b75f;
}

.c12 {
  fill: #68b75f;
}

@media print {
  .c1 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c3:hover {
    background-color: transparent;
  }
}

<div>
    <div
      class="c0 MuiBox-root"
    >
      <span
        class="c1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root c2 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root c3"
      >
        <input
          class="c4 PrivateSwitchBase-input"
          data-indeterminate="false"
          type="checkbox"
          value="testValue"
        />
        <svg
          aria-hidden="true"
          class="c5 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <div
        class="c6 MuiBox-root"
      >
        <svg
          aria-hidden="true"
          class="c5 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c7"
          data-testid="AssignmentIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
          />
        </svg>
        <div
          class="c8 MuiBox-root"
        >
          <p
            class="c9 MuiTypography-root MuiTypography-body1"
          >
            Title
          </p>
          <p
            class="c10 MuiTypography-root MuiTypography-body1"
          >
            ID: ID Last Modified: Invalid Date
          </p>
        </div>
      </div>
      <div
        class="c11 MuiBox-root"
        role="button"
      >
        <svg
          aria-hidden="true"
          class="c5 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c12"
          data-testid="CheckCircleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <p
          class="c13 MuiTypography-root MuiTypography-body1 c14"
        >
          In Library
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
